src/languages/db2i/db2i.formatter.ts

```diff
@@ -0,0 +1,193 @@
+import { DialectOptions } from '../../dialect.js';
+import { expandPhrases } from '../../expandPhrases.js';
+import { functions } from './db2i.functions.js';
+import { keywords } from './db2i.keywords.js';
+
+const reservedSelect = expandPhrases(['SELECT [ALL | DISTINCT]']);
+
+const reservedClauses = expandPhrases([
+  // queries
+  'WITH',
+  'FROM',
+  'WHERE',
+  'GROUP BY',
+  'HAVING',
+  'PARTITION BY',
+  'ORDER [SIBLINGS] BY [INPUT SEQUENCE]',
+  'OFFSET {ROW | ROWS}',
+  'FETCH {FIRST | NEXT} {ROW | ROWS} ONLY',
+  'LIMIT',
+  'FOR UPDATE [OF]',
+  'FOR READ ONLY',
+  'OPTIMIZE FOR [ALL] {ROW | ROWS}',
+  // Data modification
+  // - insert:
+  'INSERT INTO',
+  'VALUES',
+  // - update:
+  'SET',
+  // - merge:
+  'MERGE INTO',
+  'WHEN [NOT] MATCHED [THEN]',
+  'UPDATE SET',
+  'DELETE',
+  'INSERT',
+  // Data definition
+  'CREATE [OR REPLACE] TABLE [FOR SYSTEM NAME]',
+  'CREATE [OR REPLACE] [RECURSIVE] VIEW [FOR SYSTEM NAME]',
+]);
+
+const onelineClauses = expandPhrases([
+  // - update:
+  'UPDATE',
+  'WHERE CURRENT OF',
+  'WITH {NC | RR | RS | CS | UR}',
+  // - delete:
+  'DELETE FROM',
+  // - drop table:
+  'DROP TABLE',
+  // alter table:
+  'ALTER TABLE',
+  'ADD [COLUMN]',
+  'ALTER [COLUMN]',
+  'DROP [COLUMN]',
+  'RENAME [COLUMN]',
+  'SET DATA TYPE', // for alter column
+  'SET {GENERATED ALWAYS | GENERATED BY DEFAULT}', // for alter column
+  'SET NOT NULL', // for alter column
+  'SET {NOT HIDDEN | IMPLICITLY HIDDEN}', // for alter column
+  'SET FIELDPROC', // for alter column
+  'DROP {DEFAULT | NOT NULL | GENERATED | IDENTITY | ROW CHANGE TIMESTAMP | FIELDPROC}', // for alter column
+  // - truncate:
+  'TRUNCATE [TABLE]',
+  // other
+  'SET [CURRENT] SCHEMA',
+  // https://www.ibm.com/docs/en/i/7.5?topic=reference-statements
```

Comment:
Lots of entries listed below are likely incorrect. They're also incorrect in the original `DB2` implementation. This list should only contain phrases that can begin a statement or clause.

For example the phrase `SELECT INTO` can not appear in valid SQL, instead there can be something like `SELECT col1, col2 FROM tbl1 INTO tbl2`. So probably better to add an entry for `INTO`, so the `INTO`-part gets formatted on a separate line.

Reply:
Ah, thank you for your explanation and suggestion. I will go through these and change them as required.

Reply:
Issues should be fixed now - all tests run without errors.